### PR TITLE
Improve loading table

### PR DESCRIPTION
**Pull request by clostao**
Between pagination updates the table was not showing any loading spinner and just the previous page. I added a loading spinner and some little improvements.

### CODE REVIEW

File: frontend/src/components/Files/ObjectDeleteModal.tsx [hunk 1]
```diff
@@ -23,24 +23,19 @@ export const ObjectDeleteModal = ({
...
     },
     onError: (error) => {
       console.error('error', error);
     },
-    context: {
-      headers: {
-        Authorization: `Bearer ${session.data?.accessToken}`,
-      },
-    },
+    client: gqlClient,
```
Comment (marc-aurele-besner) [inline frontend/src/components/Files/ObjectDeleteModal.tsx:38]: One other approach will be to create a ApolloProvider and use it in layout to wrap your children's in the provider, so you won't need to pass in the client on each query

**Review by marc-aurele-besner — COMMENTED** [review]
I think wrapping your gqlClient in a provider will simplify all these hooks 
smt like https://github.com/autonomys/astral/blob/36cf8b444af90288d9073d311a80c94c0724fdbf/explorer/src/providers/IndexersProvider.tsx#L105

File: frontend/src/components/Files/ObjectDeleteModal.tsx (hunk 1 above)
Comment (clostao) [reply frontend/src/components/Files/ObjectDeleteModal.tsx]: Yeah, that's the best in this, I already had the ApolloProvider setup.

**Review by marc-aurele-besner — DISMISSED** [review]
LGTM

File: frontend/src/components/Files/ObjectDeleteModal.tsx (hunk 1 above)
Comment (clostao) [reply frontend/src/components/Files/ObjectDeleteModal.tsx]: Thanks for the comment btw!

**Review by marc-aurele-besner — APPROVED** [review]

**Pull request merged**